test/tests/p5.Envelope.js

```diff
@@ -0,0 +1,56 @@
+let expect = chai.expect;
+
+describe('p5.Envelop', function () {
+  let sf;
+
+  it('can be created and disposed', function () {
+    let envelope = new p5.Envelope();
+    envelope.dispose();
+  });
+
+  it('Testing setRange', function () {
+    let envelope = new p5.Envelope();
+    envelope.setRange(5, 6);
+    expect(envelope.aLevel).to.be.equal(5);
+    expect(envelope.rLevel).to.be.equal(6);
+    envelope.dispose();
+  });
+
+  it('Testing _setRampAD and setRampPercentages ', function () {
+    let envelope = new p5.Envelope();
+    envelope._setRampAD(5, 6);
+    envelope.setRampPercentages(5, 6);
+    expect(envelope._rampAttackTime).to.be.equal(5);
+    expect(envelope._rampDecayTime).to.be.equal(6);
+    expect(envelope._rampHighPercentage).to.be.equal(5);
+    expect(envelope._rampLowPercentage).to.be.equal(6);
+    expect(envelope._rampAttackTC).to.be.equal(0.27143405118953234);
+    expect(envelope._rampDecayTC).to.be.equal(600000000);
+
+    envelope.dispose();
+  });
+
+  it('Testing setExp', function () {
+    let envelope = new p5.Envelope();
+    envelope.setExp(true);
+    expect(envelope.isExponential).to.be.equal(true);
```

Comment:
we could test that `isExponential` defaults to `false` first, before calling  `setExp(true);`